### PR TITLE
Implement step and pipeline models, validation

### LINKED ISSUE #24: Add Pipelines to a Garden
_Short description_

# Assumptions:
1. 
2. 

# Acceptance Criteria
Given..., when..., then...

### PR DESCRIPTION
**Pull request by OwenPriceSkelly**
closes #24 

----
:books: Documentation preview :books:: https://garden-ai--26.org.readthedocs.build/en/26/

### CODE REVIEW

File: tests/test_garden.py [hunk 1]
```diff
@@ -236,3 +237,126 @@ def test_register_metadata(garden_client, garden_required_fields_only, tmp_path)
     with open(tmp_path / "metadata.json", "r") as f:
         json_contents = f.read()
         assert json_contents == garden.json()
+
```
Comment (BenGalewsky) [inline tests/test_garden.py:240]: Please split this up into smaller, focused python files. Any python file that starts with `test_` will be used by pytest

Comment (OwenPriceSkelly) [reply tests/test_garden.py]: done, there are now distinct `test_` files for `garden.py` and `models.py` - I've also moved all fixtures to a `conftest.py` (including the ones you'd written, happy to move those back to test_garden though if they should live in the tests file for whatever reason)

**Review by OwenPriceSkelly — COMMENTED** [review]
I didn't love having this module be pydantic_models because e.g. `from pydantic_models import Garden` seems like it might be coming from somewhere in pydantic, but also I can see that it's also likely to get confusing when we start writing code to do stuff with models (AI kind) if that code lives in the same directory as an unrelated `models.py` file. also possible I'm making a mountain out of a molehill

**Review by BenGalewsky — APPROVED** [review]
Excellent work!

**Pull request merged**